arangod/Aql/ExecutionNode.cpp

```diff
@@ -836,6 +864,18 @@ void ExecutionNode::toVelocyPackHelperGeneric(VPackBuilder& nodes, unsigned flag
       }
     }
 
+    nodes.add(VPackValue("regsToKeepStack"));
+    {
+      VPackArrayBuilder guard(&nodes);
+      //TRI_ASSERT(!_regsToKeepStack.empty()); -- can be empty in 3.6
+      for (auto const& stackEntry : getRegsToKeepStack()) {
```

Comment:
The comment one line above says that `_regsToKeepStack` could be empty.
Are we then still serializing the empty array into the `"regsToKeepStack"` VelocyPack result attribute for serialization?
If so, deserialization will fail in line 498 because of `regsToKeepStackSlice.length() == 0`.

Reply:
No. I intentionally used `getRegsToKeepStack` because it never returns an empty stack. (It lazily computes the result), hence the deserialization should always be fine.

This stunt is all just because of upgrade procedure. In 3.8 this mess can be cleaned up.

Reply:
Vale